src/v/kafka/client/direct_consumer/data_queue.cc

```diff
@@ -10,60 +10,91 @@
  */
 #include "kafka/client/direct_consumer/data_queue.h"
 
+#include "base/vassert.h"
+#include "kafka/protocol/errors.h"
+
 namespace kafka::client {
 
+namespace {
+ss::logger log{"data_queue"};
```

Comment:
i'm wondering if this logger is too fine grained? perhaps you want to take a reference to an existing logger?